### PR TITLE
Add loading indicator when submitting location in Tax task

### PR DESCRIPTION
**Pull request by ilyasfoo**
### Changes proposed in this Pull Request:

This PR is an upstream changes from https://github.com/Automattic/wc-calypso-bridge/pull/1261/files#diff-a577aa04a40191a102dfb08f7b739e30431f3b997d883de14721adcfd362c4c9R7:

1. Add a loading indicator when submitting store address in the tax task.
2. Fixes unintended reset of `isRequesting` state when updating settings
3. Add preemptive set `isRequesting` to true when optimistically updating settings to allow loading indicator

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Go to WooCommerce > Settings
2. In General tab, remove `Address line 1`, `City`, and `Postcode / ZIP`
3. Set `Country` to `Afghanistan`
4. Click on `Save changes`
5. Go to WooCommerce > Home
1. Click `Add tax rates` task in WooCommerce Home
1. Click `WooCommerce Tax`
1. Fill in the address information but ensure country is set to `Afghanistan`
2. Click on `Continue`
1. Observe `Continue` button is changed to `busy` state as per ([screenshot](https://github.com/woocommerce/woocommerce/assets/3747241/aa75094d-fc8f-412a-b81f-e5ec63ed47d3))
3. Observe when loading finishes, it automatically moves to the second

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Comment 

</details>

### CODE REVIEW

File: plugins/woocommerce-admin/client/task-lists/fills/steps/location.tsx
Comment (rjchow) [inline plugins/woocommerce-admin/client/task-lists/fills/steps/location.tsx]: Not your work, but this code boggles my mind. As I understand it, the closure will capture the reference of isSettingsError before the onSubmit runs, and thus it will probably not work as intended (?)

File: plugins/woocommerce-admin/client/task-lists/fills/steps/location.tsx
Comment (rjchow) [inline plugins/woocommerce-admin/client/task-lists/fills/steps/location.tsx]: As a result of the below comment, I wonder if we should be defensive here and wrap the updateAndPersistSettingsForGroup call in a try-catch-finally and put the setSubmitting(false) in the finally block so that if it does error, it won't just be loading forever

Comment (ilyasfoo) [reply plugins/woocommerce-admin/client/task-lists/fills/steps/location.tsx]: Good catch, @rjchow! I've moved to using try catch in https://github.com/woocommerce/woocommerce/pull/39613/commits/a585555d924f38dedf4e1c3f90e8716c61951fe3

File: plugins/woocommerce-admin/client/task-lists/fills/steps/location.tsx
Comment (ilyasfoo) [reply plugins/woocommerce-admin/client/task-lists/fills/steps/location.tsx]: Good catch, I've removed usage of `isSettingsError` altogether. I think it was pretty useless 😆 https://github.com/woocommerce/woocommerce/pull/39613/commits/a585555d924f38dedf4e1c3f90e8716c61951fe3#diff-b57e737b7d66c46d56fd34bb695f74436b4a555d3cc4ab21d0e3675178855a1aR82-R83

**Comment by rodelgc** [comment]
Hi @ilyasfoo the "Lint and Test JS" check failed. I went ahead and reran it. If it passes, I'll go ahead and merge it.

Thanks so much for the clear testing instructions and for attaching a screenshot of how the busy Continue button looks like! 

Just want to clarify something on the last step: I suppose you're referring to the "Automate taxes" step?

**Comment by rodelgc** [comment]
Update: The Lint and Test JS check still failed after a rerun. Would you be able to take a look @ilyasfoo ?

**Comment by ilyasfoo** [comment]
@rodelgc  Thanks for highlighting that! I've attempted to fix the lint issue, I'll check again soon

> Just want to clarify something on the last step: I suppose you're referring to the "Automate taxes" step?

Yes!

**Comment by rodelgc** [comment]
Thanks for addressing the lint issue @ilyasfoo! I removed the `needs: analysis` label for the meantime, since I just realized it doesn't have approval from the reviewer(s) yet. But feel free to re-add the `needs: analysis` label after PR's been approved. And we'll give a quick final look at it by then, before finally merging it.

File: plugins/woocommerce-admin/client/task-lists/fills/tax/components/store-location.tsx
Comment (chihsuan) [inline plugins/woocommerce-admin/client/task-lists/fills/tax/components/store-location.tsx:71]: Nice to see this weird logic removed. 😄

**Review by chihsuan — APPROVED** [review]
Thanks @ilyasfoo, LGTM and tested well. 👍

**Review by rjchow — APPROVED** [review]
LGTM!! Thanks for addressing the feedback 😀 
I wonder if this needs to be flagged for QA review before merging?

**Comment by nigeljamesstevenson** [comment]
> LGTM!! Thanks for addressing the feedback 😀 I wonder if this needs to be flagged for QA review before merging?

Thanks a lot for this @rjchow @ilyasfoo  and @chihsuan ! 

Please add the `needs: analysis` label and @woocommerce/solaris will take a peek before merging 😊

**Comment by ilyasfoo** [comment]
Thanks, @rjchow! Yep sounds good @rjchow @nigeljamesstevenson! I've added the label

**Pull request merged**
